test/e2e_node/cpumanager_test.go

```diff
@@ -0,0 +1,1555 @@
+ * The most important thing is to avoid long chain of beforeeach/aftereach and > 2 level of context nesting.
+ * So a **empty** context only to group labels is acceptable:
+ *   SIGDescribe
+ *     Context(label1, label2) # avoid beforeeach/aftereach and variables here
+ *       Context
+ *         It
+ *         It
+ *     Context
+ *       It
+ * Final rule of thumb: if the nesting of the context description starts to read awkward or funny or stop making sense
+ * if read as english sentence, then the nesting is likely too deep.
+ */
+
+/*
+ * About reserved CPUs:
+ * all the tests assume the first available CPUID is 0, which is pretty fair and most of time correct.
+ * A better approach would be check what we do have in the node. This is deferred to a later stage alongside
+ * other improvements.
+ */
+var _ = SIGDescribe("CPU Manager", ginkgo.Ordered, framework.WithSerial(), feature.CPUManager, func() {
+	f := framework.NewDefaultFramework("cpumanager-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	// original kubeletconfig before the context start, to be restored
+	var oldCfg *kubeletconfig.KubeletConfiguration
+	var localNode *v1.Node
+	var onlineCPUs cpuset.CPUSet
+	var smtLevel int
+	// tracks all the pods created by a It() block. Best would be a namespace per It block
+	// TODO: move to a namespace per It block?
+	var podMap map[string]*v1.Pod
+
+	ginkgo.BeforeAll(func(ctx context.Context) {
+		var err error
+		oldCfg, err = getCurrentKubeletConfig(ctx)
+		framework.ExpectNoError(err)
+
+		onlineCPUs, err = getOnlineCPUs() // this should not change at all, at least during this suite lifetime
+		framework.ExpectNoError(err)
+		framework.Logf("Online CPUs: %s", onlineCPUs)
+
+		smtLevel = smtLevelFromSysFS() // this should not change at all, at least during this suite lifetime
+		framework.Logf("SMT level %d", smtLevel)
+	})
+
+	ginkgo.AfterAll(func(ctx context.Context) {
+		updateKubeletConfig(ctx, f, oldCfg, true)
+	})
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		podMap = make(map[string]*v1.Pod)
+	})
+
+	ginkgo.AfterEach(func(ctx context.Context) {
+		deletePodsAsync(ctx, f, podMap)
+	})
+
+	ginkgo.When("running non-guaranteed pods tests", ginkgo.Label("non-guaranteed", "reserved-cpus"), func() {
+		ginkgo.It("should let the container access all the online CPUs without a reserved CPUs set", func(ctx context.Context) {
+			_ = updateKubeletConfigIfNeeded(ctx, f, configureCPUManagerInKubelet(oldCfg, &cpuManagerKubeletArguments{
+				policyName:         string(cpumanager.PolicyStatic),
+				reservedSystemCPUs: cpuset.CPUSet{},
+			}))
+
+			pod := makeCPUManagerPod("non-gu-pod", []ctnAttribute{
+				{
+					ctnName:    "non-gu-container",
+					cpuRequest: "100m",
+					cpuLimit:   "200m",
+				},
+			})
+			pod = e2epod.NewPodClient(f).CreateSync(ctx, pod)
+			podMap[string(pod.UID)] = pod
+
+			ginkgo.By("checking if the expected cpuset was assigned")
```

Comment:
Perhaps adding something like this below to reduce Flaky tests? 

```suggestion
		       ginkgo.By("ensuring the testing pod is in failed state")
		       framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name, time.Minute))
			ginkgo.By("checking if the expected cpuset was assigned")
```

Reply:
thing is this is baked into `CreateSync` already, so I think the cause is something else
```
// CreateSync creates a new pod according to the framework specifications, and wait for it to start and be running and ready.
func (c *PodClient) CreateSync(ctx context.Context, pod *v1.Pod) *v1.Pod {
	ginkgo.GinkgoHelper()
	p := c.Create(ctx, pod)
	framework.ExpectNoError(WaitTimeoutForPodReadyInNamespace(ctx, c.f.ClientSet, p.Name, c.namespace, framework.PodStartTimeout))
	// Get the newest pod after it becomes running and ready, some status may change after pod created, such as pod ip.
	p, err := c.Get(ctx, p.Name, metav1.GetOptions{})
	framework.ExpectNoError(err)
	return p
}
```

Reply:
True